pyteal/ast/jsonref.py

```diff
@@ -0,0 +1,107 @@
+from typing import TYPE_CHECKING
+from enum import Enum
+
+from pyteal.types import TealType, require_type
+from pyteal.errors import verifyFieldVersion, verifyTealVersion
+from pyteal.ir import TealOp, Op, TealBlock
+from pyteal.ast.expr import Expr
+from pyteal.ast.leafexpr import LeafExpr
+
+if TYPE_CHECKING:
+    from pyteal.compiler import CompileOptions
+
+
+class JsonRefType(Enum):
+    # fmt: off
+    #              id |    name   |      type      | min version
+    string = (0, "JSONString", TealType.bytes,  7)
+    uint64 = (1, "JSONUint64", TealType.uint64, 7)
+    object = (2, "JSONObject", TealType.bytes,  7)
+    # fmt: on
+
+    def __init__(self, id: int, name: str, type: TealType, min_version: int) -> None:
+        self.id = id
+        self.arg_name = name
+        self.ret_type = type
+        self.min_version = min_version
+
+    def type_of(self) -> TealType:
+        return self.ret_type
+
+
+JsonRefType.__module__ = "pyteal"
+
+
+class JsonRef(LeafExpr):
+    """An expression that accesses the value associated with a given key from a supported utf-8 encoded json object.
+
+    The json object must satisfy a [particular specification](https://github.com/algorand/go-algorand/blob/master/data/transactions/logic/jsonspec.md).
```

Comment:
@jdtzmn Minor:  Based on https://devguide.python.org/documenting/#external-links, I think the external link ought to be formatted differently.

```suggestion
    The json object must satisfy a `particular specification <https://github.com/algorand/go-algorand/blob/master/data/transactions/logic/jsonspec.md>`_.
```

Reply:
@jdtzmn Got it RE:  VSCode.  I'm in Pycharm, which _does_ render the link differently.

Additionally - Since API docstrings are added to https://pyteal.readthedocs.io/en/stable/api.html, you can observe the difference by generating the docs and viewing the HTML files via web browser.